src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -74,11 +74,19 @@ public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] Function
 
             bool success, configChanged;
             FunctionMetadataResponse functionMetadataResponse;
-            using (fileMonitoringService.SuspendRestart())
+            using (fileMonitoringService?.SuspendRestart())
             {
                 (success, configChanged, functionMetadataResponse) = await _functionsManager.CreateOrUpdate(name, functionMetadata, Request);
             }
-            await fileMonitoringService.ScheduleRestartAsync(false);
+            Task restartTask = null;
```

Comment:
Can you please introduce a method to encapsulate this pattern so we don't have to repeat it?